### DISCUSSION TITLE
Cannot add translated text in TextBlock

### DISCUSSION THREAD

**Discussion by Taliesin60** [AvaloniaUI/Avalonia#12164] [Q&A] [answered]
**Describe the bug**
I want to attach a translated resource to a textblock in new Avalonia App and have error :  [AVLN:0004] Unable to resolve Login as static field, property, constant or enum value Line 14, position 20.

**To Reproduce**

1. Create a Cross Platform Avalonia App
2. Create a Resx file to translate the app
3. In the View, add a textblock with Text = {x:Static Resources.Item}
4. In compilation, have the error "[AVLN:0004] Unable to resolve Login as static field, property, constant or enum value Line 14, position 20."

**Expected behavior**
Set text value of textblock as translated resource value

**Desktop:**

- OS: Windows 11
- Version [e.g. 0.10.0-rc1 or 0.9.12]

**Additional context**
using Avalonia 11 nuget

**Comment by robloo** [comment]
Please do a search for other methods of localization supported in Avalonia. There are 2-3 good options.

.resx files are not automatically parsed and used to generate c# code that can be referenced elsewhere. There is a way to do this though.

Additionally, you might not want static lookup if you want to change the UI language at runtime.

I believe this issue should be closed for now.

**Comment by maxkatz6** [comment]
Resx and xStatic based localization works fine and there is an issue in your code somewhere else. Need more information.

**Comment by timunie** [comment]
I guess one needs to make the source gen public. At least in WPF it was possible. For some reason, it's internal by default.

**Comment by robloo** [comment]
Since when did Avalonia get a source generator for this!?

**Comment by timunie** [comment]
Not avalonia but resxmanager in VS does have a generator for it. Maybe source gen was wrong wording here. Would be cool to have one if I think about it 🤔

**Comment by IanRawley** [comment]
Did you include a namespace for the resx files?

xmlns:L="clr-namespace:location.of.resx"
HeaderText="{x:Static L:AppStrings.Label_Name}" 

works fine for me, where AppStrings is the base name of a resx file.

**Comment by rabbitism** [comment]
iirc, the issue comes because the generated C# code is internal but not public, and it is not accessible in XAML. you need to set the visibility to public in visual studio, or set it in csproj

```xml
    <ItemGroup>
      <EmbeddedResource Update="Properties\Resources.resx">
        <Generator>PublicResXFileCodeGenerator</Generator> 
        <LastGenOutput>Resources.Designer.cs</LastGenOutput>
      </EmbeddedResource>
    </ItemGroup>
```

**Comment by zsuatem** [comment, marked as answer]
I have a very similar problem. :/

In how-to-do guides I found "Localizing using ResX" and I wanted to try this method but for some reason it doesn't want to work.
It always ends with the error "Unable to resolve xyz as static field, property, constant or enum value Line x, position y.".
For this reason I downloaded https://github.com/AvaloniaUI/AvaloniaUI.QuickGuides and here the Localization project works however if I add my own resx it no longer wants to work unfortunately.

Added new resx in Localization - does not work
![image](https://github.com/AvaloniaUI/Avalonia/assets/17302628/d811518b-6872-49ef-bd81-09c3c3562849)

csproj
```xml
    <ItemGroup>
      <AvaloniaResource Update="Assets\Resource1.Designer.cs">
        <DesignTime>True</DesignTime>
        <AutoGen>True</AutoGen>
        <DependentUpon>Resource1.resx</DependentUpon>
      </AvaloniaResource>
      <AvaloniaResource Update="Assets\Resource1.resx">
        <Generator>PublicResXFileCodeGenerator</Generator>
        <LastGenOutput>Resource1.Designer.cs</LastGenOutput>
      </AvaloniaResource>
    </ItemGroup>
```
View
```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:Localization.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:assets="clr-namespace:Localization.Assets"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="Localization.Views.MainWindow"
        x:DataType="vm:MainWindowViewModel"
        Icon="/Assets/avalonia-logo.ico"
        Title="MakingLists">

    <Design.DataContext>
        <!-- This only sets the DataContext for the previewer in an IDE,
             to set the actual DataContext for runtime, set the DataContext property in code (look at App.axaml.cs) -->
        <vm:MainWindowViewModel />
    </Design.DataContext>
    <Panel>
        <TextBlock Text="{x:Static assets:Resources.GreetingText}"/>
		
		<!-- Not work-->
		<TextBlock Text="{x:Static assets:Resource1.GreetingText}"/>
    </Panel>
</Window>
```

Project
[AvaloniaUI.QuickGuides.zip](https://github.com/AvaloniaUI/Avalonia/files/12131529/AvaloniaUI.QuickGuides.zip)

**Reply by timunie** [reply]
If you don't place it into assets but another folder, does it work then? I have the feeling that Assets has special build instructions

**Reply by zsuatem** [reply]
It seems to be working, thanks

**Comment by sunwayking** [comment]
Make sure the "Access Modifier" in main file Resources.resx is "Public", not the default value "Internal" in Visual Studio. Then you would see no error during compiling your project.
